packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -152,7 +152,7 @@ abstract class Target {
   ///
   /// Returning `true` will cause [build] to be skipped. This is equivalent
   /// to a build that produces no outputs.
-  bool canSkip(Environment environment) => false;
+  Future<bool> canSkip(Environment environment) async => false;
```

Comment:
Why does this need to be made async?

Reply:
I have a big PR coming and I'm trying to break it out into smaller bits. Eventually I want to add this:

https://github.com/vashworth/flutter/blob/f515bd0f507480566bf873330e67df11b2688f94/packages/flutter_tools/lib/src/build_system/targets/darwin.dart#L123-L171